containers.go

```diff
@@ -26,7 +26,7 @@ func (c *Container) Clone(ctx context.Context, params *ContainerCloneOptions) (n
 	if err := c.client.Post(ctx, fmt.Sprintf("/nodes/%s/lxc/%d/clone", c.Node, c.VMID), params, &upid); err != nil {
 		return 0, nil, err
 	}
-	return newid, NewTask(upid, c.client), nil
+	return params.NewID, NewTask(upid, c.client), nil
```

Comment:
could you double check, instead of returning params.NewID here it looks the bug seems to be a `:=` at line 20 creating a new variable instead of reusing the one defined in the return. the virtual machines implementation does not have the colon so it's possible that one was already working?

Reply:
I actually found the bug in the virtual machine implementation. The problem is that `newid` is never assigned a value when `params.NewID != 0`. I didn't realise that `newid` is a named return value. I think I improved the PR by using the named return values and assigning `params.NewID` to `newid` if it isn't `0`.